src/EFCore.MySql/Storage/Internal/ByteArrayComparer.cs

```diff
@@ -0,0 +1,20 @@
+﻿// Copyright (c) Pomelo Foundation. All rights reserved.
+// Licensed under the MIT. See LICENSE in the project root for license information.
+
+using System.Collections;
+using System.Linq;
+using Microsoft.EntityFrameworkCore.ChangeTracking;
+
+namespace EFCore.MySql.Storage.Internal
+{
+    public class ByteArrayComparer : ValueComparer<byte[]>

```

Comment:
Why is this needed?

Reply:
To compare the binary representations of `timestamp`. Is there a better way of doing this?

Reply:
Just create a ValueComparer instance passing in "favorStructuralComparisons" as true.

Reply:
Never mind. I see now that you need to snapshot it as well.